### PR TITLE
fix(vector source,opentelemetry source): emit errors on grpc errors

### PR DESCRIPTION
**Pull request by StephenWakely**
Ref #14501 

This emits errors when an error occurs in the grpc server used by the Vector and OpenTelemetry sources.

Signed-off-by: Stephen Wakely <fungus.humungus@gmail.com>

### CODE REVIEW

File: src/internal_events/grpc.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+impl<E> InternalEvent for GrpcError<E>
+where
+    E: std::fmt::Display,
+{
+    fn emit(self) {
+        error!(message = "Grpc error.",
+               error = %self.error,
+               error_type = error_type::REQUEST_FAILED,
+               stage = error_stage::RECEIVING,
+               internal_log_rate_secs = 10
```
Comment (neuronull) [inline src/internal_events/grpc.rs:40]:
```suggestion
               internal_log_rate_limit = true,
```

File: src/internal_events/grpc.rs [hunk 2]
```diff
@@ -9,11 +9,12 @@ pub struct GrpcInvalidCompressionScheme<'a> {
...
-               error = ?self.status.message(),
-               error_type = error_type::REQUEST_FAILED,
-               stage = error_stage::RECEIVING,
-               internal_log_rate_secs = 10
+        error!(
+            message = "Invalid compression scheme.",
+            error = ?self.status.message(),
+            error_type = error_type::REQUEST_FAILED,
+            stage = error_stage::RECEIVING,
+            internal_log_rate_secs = true
```
Comment (neuronull) [inline src/internal_events/grpc.rs:17]: Almost :) The name was changed recently.

```suggestion
            internal_log_rate_limit = true
```

File: src/internal_events/grpc.rs [hunk 3]
```diff
@@ -33,11 +34,12 @@ where
...
-               error = %self.error,
-               error_type = error_type::REQUEST_FAILED,
-               stage = error_stage::RECEIVING,
-               internal_log_rate_secs = 10
+        error!(
+            message = "Grpc error.",
+            error = %self.error,
+            error_type = error_type::REQUEST_FAILED,
+            stage = error_stage::RECEIVING,
+            internal_log_rate_secs = true
```
Comment (neuronull) [inline src/internal_events/grpc.rs:42]:
```suggestion
            internal_log_rate_limit = true
```

File: src/internal_events/grpc.rs (hunk 2 above)
Comment (StephenWakely) [reply src/internal_events/grpc.rs:17]: darn it

**Review by spencergilbert — APPROVED** [review]
🤔 does the title actually pass the semantic check? shouldn't it be like `fix(vector source,opentelemetry source): emit errors on grpc errors`?

**Comment by StephenWakely** [comment]
> does the title actually pass the semantic check? shouldn't it be like `fix(vector source,opentelemetry source): emit errors on grpc errors`?

Good catch. That check doesn't seem to be running..

**Pull request merged**